contracts/interfaces/bridge/IStateReceiver.sol

```diff
@@ -0,0 +1,11 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.19;
+
+interface IStateReceiver {
+    /**
+     * @notice Called by gateway when state is received from source chain
+     * @param sender Address of the sender on the child chain
+     * @param data Data sent by the sender
+     */
+    function onStateReceive(uint256 id, address sender, bytes calldata data) external;
```

Comment:
WDYT (just don't forget to change the string under quotas in the Gateway, where this function gets invoked).
```suggestion
    function onMsgReceived(uint256 id, address sender, bytes calldata data) external;
```

Reply:
We will address this in another task.